### PR TITLE
check lints and code formatting in CI

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**

### CODE REVIEW

**Comment by ChristianBeilschmidt** [comment]
[Linter für Python Code](https://trello.com/c/uGYufWwB/247-linter-f%C3%BCr-python-code)

**Review by michaelmattig — CHANGES_REQUESTED** [review]

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,7 @@
 build
 flufl.lock # for cartopy mpl tests
+pycodestyle # formatter
```
[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply requirements.txt:3]: These are the dev dependencies. The install requirements are in setup.cfg

File: geoengine/datasets.py [hunk 2]
```diff
@@ -94,13 +115,17 @@ def to_dict(self) -> Dict[str, str]:
...
+    '''An open-ended time duration'''
+
     def to_dict(self) -> Dict[str, str]:
         return {
             "type": "infinite",
         }
 
 
 class OgrSourceDatasetTimeType:
+    '''A time type specifiction for OGR dataset definitions'''
```
[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply geoengine/datasets.py]: 145129a

File: README.md [hunk 3]
```diff
@@ -40,6 +40,30 @@ python3 -m pip install --upgrade build
...
+
+## Lints
+
+Our CI automatically checks for lint errors.
+We use `pylint` to check the code.
+You can check it by calling:
+
+```bash
+python3 -m geoengine
+python3 -m tests
```
[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply README.md]: 145129a

File: README.md [hunk 4]
```diff
@@ -40,6 +40,30 @@ python3 -m pip install --upgrade build
...
+Our tip is to install `autopep8` and use it to format the code.
+
+## Lints
+
+Our CI automatically checks for lint errors.
+We use `pylint` to check the code.
+You can check it by calling:
+
+```bash
+python3 -m geoengine
```
[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply README.md]: 145129a

**Review by michaelmattig — APPROVED** [review]

**Pull request merged**